dot/sync/message_test.go

```diff
@@ -0,0 +1,439 @@
+// Copyright 2021 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package sync
+
+import (
+	"errors"
+	"reflect"
+	"testing"
+
+	"github.com/ChainSafe/gossamer/dot/network"
+	"github.com/ChainSafe/gossamer/dot/types"
+	"github.com/ChainSafe/gossamer/lib/common"
+	"github.com/ChainSafe/gossamer/lib/common/variadic"
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestService_CreateBlockResponse(t *testing.T) {
+	t.Parallel()
+
+	ctrl := gomock.NewController(t)
+
+	mockBlockState := NewMockBlockState(ctrl)
+	mockBlockState.EXPECT().BestBlockNumber().Return(uint(1), nil).Times(8)
+	mockBlockState.EXPECT().GetHashByNumber(gomock.Any()).DoAndReturn(func(uint) (
+		common.Hash, error) {
+		return common.Hash{1, 2}, nil
+	}).Times(5)
+	mockBlockState.EXPECT().IsDescendantOf(gomock.AssignableToTypeOf(common.Hash{}),
+		gomock.AssignableToTypeOf(common.Hash{})).Return(true, nil).Times(4)
+	mockBlockState.EXPECT().GetHeader(gomock.AssignableToTypeOf(common.Hash{})).Return(&types.Header{
+		Number: 1,
+	}, nil).Times(4)
+	mockBlockState.EXPECT().SubChain(gomock.AssignableToTypeOf(common.Hash{}),
+		gomock.AssignableToTypeOf(common.Hash{})).Return([]common.Hash{{1, 2}}, nil).Times(4)
+	mockBlockState.EXPECT().GetHeaderByNumber(uint(1)).Return(&types.Header{
+		Number: 1,
+	}, nil)
```

Comment:
Configure these with functional test case fields in each subtest

Reply:
Updated.